### PR TITLE
erl_interface warning fixes

### PR DESCRIPTION
**Pull request by frej**

### CODE REVIEW

File: erts/etc/common/escript.c [hunk 1]
```diff
@@ -286,7 +286,7 @@ find_prog(char *origpath)
                     beg = end + 1;
                     continue;
                 }
-                strncpy(dir, beg, sz);
+                mempcpy(dir, beg, sz);
```
Comment (rickard-green) [inline erts/etc/common/escript.c:289]:
```suggestion
                memcpy(dir, beg, sz);
```
After googleing a bit is seems that `mempcpy()` have portability issues and `memcpy()` is identical if the return value is ignored.

Comment (frej) [reply erts/etc/common/escript.c]: Considering that `mempcpy` is a typo on my part, I 100% agree that `memcpy` is what it should be, do you want a refreshed patch @rickard-green ?

Comment (rickard-green) [reply erts/etc/common/escript.c]: You can squash the commit that I added into your commit.

Comment (frej) [reply erts/etc/common/escript.c]: Did that and refreshed the patches to current master. I also fixed some errors in the commit message, I wonder what I was thinking when I wrote it, the fix we're discussing is to escript and not erl_interface as the commit message claimed. :)

Comment (rickard-green) [reply erts/etc/common/escript.c]: 👍

**Pull request merged**